### ISSUE TITLE
Update to 1.11.2?

### ISSUE THREAD

**Issue by polyzium** [ata4/mineshot#17]
As described in the title.

**Comment by MggMuggins** [comment]
Does the 1.10.2 version run in 1.11.2? Try it and get back to me.

**Comment by polyzium** [comment]
Although Minecraft runs fine when i put the built .jar into mods folder, the mod doesn't work. It just doesn't show up in the mod list

**Comment by MggMuggins** [comment]
Does it have anything to say in the mod list?

**Comment by polyzium** [comment]
I guess it should show up in the mod list as a Forge mod, i didn't try this out in 1.10 yet.
Also here's the error i found in the log which makes Mineshot not to work:
```
[16:44:39] [main/ERROR] [FML]: The coremod info.ata4.minecraft.mineshot.MineshotCore is requesting minecraft version 1.10.2 and minecraft is 1.11.2. It will be ignored.
```
I also tried changing build.gradle to match 1.11.2 - it failed to build.

**Comment by MggMuggins** [comment]
That makes sense. Let me get a gradle environment set up with Mineshot if I don't already and I'll see if I can get it to run.

**Comment by MggMuggins** [comment]
Make sure to change the MC version annotation in MineshotCore.java. Then try.

**Comment by polyzium** [comment]
I changed the MCVersion annotation both in MineshotCore.java and build.gradle, it falied to build. But if i leave build.gradle to target 1.10.2, then with this build, MC 1.11.2 still loads but it doesn't load the mod and thus it doesn't show up in the mod list. In the log it said the following:
```
[19:58:05] [main/DEBUG] [FML/]: Examining for coremod candidacy Mineshot-v1.71.jar
[19:58:05] [main/TRACE] [FML/]: Found FMLCorePluginContainsFMLMod marker in Mineshot-v1.71.jar, it will be examined later for regular @Mod instances
[19:58:05] [main/DEBUG] [FML/]: Instantiating coremod class MineshotCore
[19:58:05] [main/DEBUG] [FML/]: The coremod info.ata4.minecraft.mineshot.MineshotCore requested minecraft version 1.11.2 and minecraft is 1.11.2. It will be loaded.
[19:58:05] [main/DEBUG] [FML/]: Enqueued coremod MineshotCore
```
later...
```
[19:58:07] [main/DEBUG] [FML/]: Injecting coremod MineshotCore {info.ata4.minecraft.mineshot.MineshotCore} class transformers
[19:58:07] [main/TRACE] [FML/]: Registering transformer info.ata4.minecraft.mineshot.client.transform.EntityRendererTransformer
[19:58:08] [main/TRACE] [FML/]: Registering transformer info.ata4.minecraft.mineshot.client.transform.FrustumTransformer
[19:58:08] [main/DEBUG] [FML/]: Injection complete
[19:58:08] [main/DEBUG] [FML/]: Running coremod plugin for MineshotCore {info.ata4.minecraft.mineshot.MineshotCore}
[19:58:08] [main/DEBUG] [FML/]: Running coremod plugin MineshotCore
[19:58:08] [main/DEBUG] [FML/]: Coremod plugin class MineshotCore run successfully
```
later...
```
[19:58:21] [Client thread/DEBUG] [FML/]: Examining file Mineshot-v1.71.jar for potential mods
[19:58:21] [Client thread/TRACE] [FML/]: Located mcmod.info file in file Mineshot-v1.71.jar
[19:58:21] [Client thread/DEBUG] [FML/]: Identified a mod of type Lnet/minecraftforge/fml/common/Mod; (info.ata4.minecraft.mineshot.Mineshot) - loading
[19:58:21] [Client thread/WARN] [FML/]: ****************************************
[19:58:21] [Client thread/WARN] [FML/]: * The modid Mineshot is not the same as it's lowercase version. Lowercasing is enforced in 1.11
[19:58:21] [Client thread/WARN] [FML/]: *  at net.minecraftforge.fml.common.FMLModContainer.sanityCheckModId(FMLModContainer.java:147)
[19:58:21] [Client thread/WARN] [FML/]: *  at net.minecraftforge.fml.common.FMLModContainer.<init>(FMLModContainer.java:130)
[19:58:21] [Client thread/WARN] [FML/]: *  at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[19:58:21] [Client thread/WARN] [FML/]: *  at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
[19:58:21] [Client thread/WARN] [FML/]: *  at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[19:58:21] [Client thread/WARN] [FML/]: *  at java.lang.reflect.Constructor.newInstance(Constructor.java:423)...
[19:58:21] [Client thread/WARN] [FML/]: ****************************************
java.lang.reflect.InvocationTargetException
        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:1.8.0_121]
        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:1.8.0_121]
        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:1.8.0_121]
        at java.lang.reflect.Constructor.newInstance(Constructor.java:423) ~[?:1.8.0_121]
        at net.minecraftforge.fml.common.ModContainerFactory.build(ModContainerFactory.java:90) [ModContainerFactory.class:?]
        at net.minecraftforge.fml.common.discovery.JarDiscoverer.discover(JarDiscoverer.java:106) [JarDiscoverer.class:?]
        at net.minecraftforge.fml.common.discovery.ContainerType.findMods(ContainerType.java:49) [ContainerType.class:?]
        at net.minecraftforge.fml.common.discovery.ModCandidate.explore(ModCandidate.java:78) [ModCandidate.class:?]
        at net.minecraftforge.fml.common.discovery.ModDiscoverer.identifyMods(ModDiscoverer.java:141) [ModDiscoverer.class:?]
        at net.minecraftforge.fml.common.Loader.identifyMods(Loader.java:414) [Loader.class:?]
        at net.minecraftforge.fml.common.Loader.loadMods(Loader.java:554) [Loader.class:?]
        at net.minecraftforge.fml.client.FMLClientHandler.beginMinecraftLoading(FMLClientHandler.java:233) [FMLClientHandler.class:?]
        at net.minecraft.client.Minecraft.func_71384_a(Minecraft.java:440) [bes.class:?]
        at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:352) [bes.class:?]
        at net.minecraft.client.main.Main.main(SourceFile:124) [Main.class:?]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_121]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_121]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_121]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_121]
        at net.minecraft.launchwrapper.Launch.launch(Launch.java:135) [launchwrapper-1.12.jar:?]
        at net.minecraft.launchwrapper.Launch.main(Launch.java:28) [launchwrapper-1.12.jar:?]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_121]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_121]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_121]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_121]
        at net.minecraft.launchwrapper.Launch.launch(Launch.java:135) [launchwrapper-1.12.jar:?]
        at net.minecraft.launchwrapper.Launch.main(Launch.java:28) [launchwrapper-1.12.jar:?]
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_121]
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_121]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_121]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_121]
        at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:236) [NewLaunch.jar:?]
        at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:297) [NewLaunch.jar:?]
        at org.multimc.EntryPoint.listen(EntryPoint.java:162) [NewLaunch.jar:?]
        at org.multimc.EntryPoint.main(EntryPoint.java:53) [NewLaunch.jar:?]
Caused by: java.lang.IllegalArgumentException: The modid Mineshot is not the same as it's lowercase version. Lowercasing will be enforced in 1.11
        at net.minecraftforge.fml.common.FMLModContainer.sanityCheckModId(FMLModContainer.java:148) ~[FMLModContainer.class:?]
        at net.minecraftforge.fml.common.FMLModContainer.<init>(FMLModContainer.java:130) ~[FMLModContainer.class:?]
        ... 29 more
```
another line
```
[19:58:23] [Client thread/INFO] [FML/]: FML has found a non-mod file Mineshot-v1.71.jar in your mods directory. It will now be injected into your classpath. This could severe stability issues, it should be removed if possible.
```
I have no idea what that lowercase enforcement is that all about...

**Comment by Kosti1** [comment]
I can fix that

**Comment by polyzium** [comment]
Put a pull request then

**Comment by Kosti1** [comment]
Ill just fork it and send a link to it

**Comment by SXRWahrheit** [comment]
@Kosti1, could you tag me when you update it? :)

**Comment by Kosti1** [comment]
@SXRWahrheit @polyzium https://github.com/Kosti1/mineshot/releases/tag/1.11.2 here you go, I built it and released

**Comment by polyzium** [comment]
It crashed.
```
[22:15:46] [Client thread/ERROR] [FML]: The following problems were captured during this phase
[22:15:46] [Client thread/ERROR] [FML]: Caught exception from mineshot (mineshot)
java.lang.NoSuchMethodError: net.minecraft.client.resources.I18n.format(Ljava/lang/String;[Ljava/lang/Object;)Ljava/lang/String;
```

**Comment by Kosti1** [comment]
Ill fix it, maybe I just have to update forge version

**Comment by polyzium** [comment]
Apparently there's no fix... all hope is on @ata4 if he will ever come back

**Comment by SXRWahrheit** [comment]
Could you elaborate on why you think it can't be fixed @polyzium?

**Comment by Kosti1** [comment]
So I am currently testing a build that I hope fixes the problem @polyzium and @SXRWahrheit

**Comment by MggMuggins** [comment]
If you guys get it fixed, go ahead and open a PR anyway. I think I spoke to ata4 at one point here...

**Comment by DentedHarp90041** [comment]
Updated to 1.12 (although 1.11.2 requested) :) https://github.com/DentedHarp90041/mineshot/releases/tag/1.12

**Comment by MggMuggins** [comment]
You should squash your commits and open a PR.

**Comment by Kosti1** [comment]
I got the bug fixed, also ported to 1.12 and 1.11.2.
made PRs and added releases to my fork

**Comment by Kosti1** [comment]
could you please close the issue

**Comment by polyzium** [comment]
Let me test if it works

**Comment by polyzium** [comment]
It is working. Closing

**Comment by ata4** [comment]
I have just ported Mineshot v1.71 myself and builds are now officially available for 1.11.2 and 1.12. Sorry for taking *that* long to get the job done, but Minecraft modding has fallen quite out of favor to me, so... yeah.
